crates/bevy_gltf/src/loader.rs

```diff
@@ -291,36 +289,174 @@ async fn load_gltf<'a, 'b, 'c>(
                     return Err(GltfError::MissingAnimationSampler(animation.index()));
                 };
 
-                let keyframes = if let Some(outputs) = reader.read_outputs() {
+                if keyframe_timestamps.is_empty() {
+                    warn!("Tried to load animation with no keyframe timestamps");
+                    continue;
+                }
+
+                let maybe_curve: Option<VariableCurve> = if let Some(outputs) =
```

Comment:
I feel this has a lot of cyclomatic complexity - if we do

```rust
let Some(outputs) = reader.read_outputs() else {
    warn!("Animations without a sampler output are not supported");
    return Err(GltfError::MissingAnimationSampler(animation.index()));
};
```

 we could achieve the same thing but with more readability? 
The same review comment applies to the following `let ... = if let Some(...) = ...` passages.